demos/http/http_demo_s3_download/http_demo_s3_download.c

```diff
@@ -0,0 +1,625 @@
+ * HTTP response headers and body.
+ *
+ * @note This demo shows how the same buffer can be re-used for storing the HTTP
+ * response after the HTTP request is sent out. Here, a separate buffer is used
+ * to download the file from the server. However, the user can decide how to use
+ * buffers to store HTTP requests and responses.
+ */
+static uint8_t userBuffer[ USER_BUFFER_LENGTH ];
+
+/**
+ * @brief A buffer used in the demo for downloading the file from the server
+ * in chunks.
+ */
+static uint8_t fileDownloadBuffer[ FILE_BUFFER_LENGTH ];
+
+/**
+ * @brief Represents header data that will be sent in an HTTP request.
+ */
+static HTTPRequestHeaders_t requestHeaders;
+
+/**
+ * @brief Configurations of the initial request headers that are passed to
+ * #HTTPClient_InitializeRequestHeaders.
+ */
+static HTTPRequestInfo_t requestInfo;
+
+/**
+ * @brief Represents a response returned from an HTTP server.
+ */
+static HTTPResponse_t response;
+
+/* The location of the host address within string S3_PRESIGNED_URL. */
+static const char * pAddress = NULL;
+
+/* The host address string extracted from S3_PRESIGNED_URL. */
+static char serverHost;
+
+/* The length of the host address found in string S3_PRESIGNED_URL. */
+static size_t serverHostLength = 0;
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Connect to HTTP server with reconnection retries.
+ *
+ * @param[out] pNetworkContext The output parameter to return the created network context.
+ *
+ * @return EXIT_FAILURE on failure; EXIT_SUCCESS on successful connection.
+ */
+static int32_t connectToServer( NetworkContext_t * pNetworkContext );
+
+/**
+ * @brief Retrieve the size of the S3 object that is specified in pPath.
+ *
+ * @param[out] pFileSize - The size of the S3 object.
+ * @param[in] pTransportInterface The transport interface for making network calls.
+ * @param[in] pHost The host name of the server.
+ * @param[in] pMethod The HTTP request method.
+ * @param[in] pPath The Request-URI to the objects of interest.
+ */
+static HTTPStatus_t getS3ObjectFileSize( size_t * pFileSize,
+                                         const TransportInterface_t * pTransportInterface,
+                                         const char * pHost,
+                                         const char * pMethod,
+                                         const char * pPath );
+
+/**
+ * @brief Send an HTTP request based on a specified method and path, then
+ * print the response received from the server.
+ *
+ * @param[in] pTransportInterface The transport interface for making network calls.
+ * @param[in] pHost The host name of the server.
+ * @param[in] pMethod The HTTP request method.
+ * @param[in] pPath The Request-URI to the objects of interest.
+ *
+ * @return EXIT_FAILURE on failure; EXIT_SUCCESS on success.
+ */
+static int downloadS3ObjectFile( const TransportInterface_t * pTransportInterface,
```

Comment:
Can we please return a bool or HTTPStatus_t here instead of EXIT_SUCCESS or EXIT_FAILURE?